File: supply-chain-management - Copy/src/app/util/notify.util.ts
import { ApiResponse } from "./api.response";

export class NotifyUtil {

    static success(response: ApiResponse): void {
      alert(response?.message || 'Successful');
    }
  
    static error(error: any): void {
      console.log(error);
      if (error.message) {
        alert(error.message);
      } else {
        alert(error.error?.message || `An error occurred`);
      }
    }
  
  }

File: supply_chain_flutter/lib/util/notify_util.dart
import 'package:flutter/material.dart';

class NotifyUtil {
  static void success(BuildContext context, String? message) {
    final msg = message ?? 'Successful';
    _showSnackBar(context, msg, Colors.green);
  }

  static void error(BuildContext context, dynamic error) {
    String errorMessage;

    if (error is String) {
      errorMessage = error;
    } else if (error is Exception) {
      errorMessage = error.toString();
    } else {
      errorMessage = error?.message ?? error.toString() ?? 'An error occurred';
    }

    _showSnackBar(context, errorMessage, Colors.red);
  }

  static void _showSnackBar(BuildContext context, String message, Color color) {
    final snackBar = SnackBar(
      content: Text(message),
      backgroundColor: color,
    );
    ScaffoldMessenger.of(context).showSnackBar(snackBar);
  }
}
